torch/_inductor/cpu_vec_isa.py

```diff
@@ -239,6 +260,23 @@ def __bool__(self) -> bool:
                 return True
         return False
 
+    # check amx_fp16 separately since it is not always supported when amx is supported
+    # amx_fp16 instrinsic compilation need gcc >=13 on platforms which support amx_fp16
+    @functools.lru_cache(None)  # noqa: B019
+    def is_amx_fp16_supported(self) -> bool:
+        if super().__bool__():
+            if config.is_fbcode():
+                return False
+            if torch.cpu._init_amx() and torch.cpu._is_amx_fp16_supported():
+                self._arch_flags += " -mamx-fp16"
+                if self.check_build(VecAMX._amx_fp16_code):
+                    return True
+                else:
+                    self._arch_flags = (
+                        VecAVX512._arch_flags + " -mamx-tile -mamx-bf16 -mamx-int8"
```

Comment:
Save the original flag and restore it here?

Reply:
I don't think it is a good practice to mutate member fields in a getter function like `is_amx_fp16_supported`. Perhaps moving this to the constructor?

Reply:
> Perhaps moving this to the constructor?

I don't quite understand ‘moving this to the constructor’, could you please give me more details? Thanks a lot.

Reply:
Add a flag `_is_amx_fp16_supported` and override `build_arch_flags` to return different `_arch_flags` based on `_is_amx_fp16_supported`.